XCUITests/TranslationSnackBarTest.swift

```diff
@@ -5,15 +5,41 @@
 import XCTest
 
 class TranslationSnackBarTest: BaseTestCase {
-    func testSnackBar() {
+    
+    // This test checks to the correct functionalty of the Translation prompt and Translation is done corrrectly using Google
+    func testSnackBarDisplayed() {
         userState.localeIsExpectedDifferent = true
         navigator.openURL(path(forTestPage: "manifesto-zh-CN.html"))
         waitUntilPageLoad()
-        navigator.goto(TranslatePageMenu)
         waitForExistence(app.buttons["TranslationPrompt.doTranslate"])
+        app.buttons["TranslationPrompt.dontTranslate"].tap()
+        XCTAssertFalse(app.buttons["TranslationPrompt.dontTranslate"].exists)
+        navigator.performAction(Action.ReloadURL)
+        waitForExistence(app.buttons["TranslationPrompt.doTranslate"])
+        navigator.performAction(Action.SelectTranslateThisPage)
+        waitForValueContains(app.textFields["url"], value: "translate.google")
+        XCUIApplication().buttons["Reload"].tap()
     }
-
-    func testSetting() {
+    
+    // This test checks to see if Translation is enabled by default from the Settings menu and can be correctly disabled
+    func testTranslationDisabled() {
         navigator.goto(TranslationSettings)
+        let translationSwitch = app.switches["TranslateSwitchValue"]
+        XCTAssertTrue(translationSwitch.isEnabled)
+        translationSwitch.tap()
+        navigator.openURL(path(forTestPage: "manifesto-zh-CN.html"))
+        waitUntilPageLoad()
+        XCTAssertFalse(app.buttons["TranslationPrompt.dontTranslate"].exists)
+    }
+    
+    // This test checks to see if Translation is correctly done when using Bing
+    func testTranslateBing() {
+        navigator.goto(TranslationSettings)
+        app.tables["Translate.Setting.Options"].staticTexts["Bing"].tap()
```

Comment:
I would also add an action for selecting the different options in the [TranslationSetting screen](https://github.com/mozilla-mobile/firefox-ios/blob/71fd6d0e53ea4fdf72a3bae50ac701f76639bf27/XCUITests/FxScreenGraph.swift#L550)
Also, please check if Bing has an id to use better than its String or use the app.tables.element(boundBy: x), where x is the order in the list (0,1,2...). Whenever is possible is better to use ids or elements identifiers not based on the string since that could change. Add a comment to say selecting Bing if that's not clear using its id or order in the table.